diff --git a/clang-tidy/CMakeLists.txt b/clang-tidy/CMakeLists.txt
index af10609..1e92def 100644
--- a/clang-tidy/CMakeLists.txt
+++ b/clang-tidy/CMakeLists.txt
@@ -34,3 +34,4 @@ add_subdirectory(misc)
 add_subdirectory(modernize)
 add_subdirectory(readability)
 add_subdirectory(utils)
+add_subdirectory(misra)
diff --git a/clang-tidy/tool/CMakeLists.txt b/clang-tidy/tool/CMakeLists.txt
index cc58148..e440252 100644
--- a/clang-tidy/tool/CMakeLists.txt
+++ b/clang-tidy/tool/CMakeLists.txt
@@ -13,6 +13,7 @@ target_link_libraries(clang-tidy
   clangTidyCERTModule
   clangTidyCppCoreGuidelinesModule
   clangTidyGoogleModule
+  clangTidyMISRAModule
   clangTidyLLVMModule
   clangTidyMiscModule
   clangTidyModernizeModule
diff --git a/clang-tidy/tool/ClangTidyMain.cpp b/clang-tidy/tool/ClangTidyMain.cpp
index 9dc6824..e0b680f 100644
--- a/clang-tidy/tool/ClangTidyMain.cpp
+++ b/clang-tidy/tool/ClangTidyMain.cpp
@@ -382,6 +382,10 @@ extern volatile int ReadabilityModuleAnchorSource;
 static int LLVM_ATTRIBUTE_UNUSED ReadabilityModuleAnchorDestination =
     ReadabilityModuleAnchorSource;
 
+// This anchor is used to force the linker to link the MISRAModule.
+extern volatile int MISRAModuleAnchorSource;
+static int LLVM_ATTRIBUTE_UNUSED MISRAModuleAnchorDestination = MISRAModuleAnchorSource;
+
 } // namespace tidy
 } // namespace clang
 
