diff --git a/clang-tidy/CMakeLists.txt b/clang-tidy/CMakeLists.txt
index d08d56da..3800d8d4 100644
--- a/clang-tidy/CMakeLists.txt
+++ b/clang-tidy/CMakeLists.txt
@@ -40,3 +40,4 @@ add_subdirectory(performance)
 add_subdirectory(readability)
 add_subdirectory(safety)
 add_subdirectory(utils)
+add_subdirectory(misra)
diff --git a/clang-tidy/tool/CMakeLists.txt b/clang-tidy/tool/CMakeLists.txt
index 77222117..fa114e78 100644
--- a/clang-tidy/tool/CMakeLists.txt
+++ b/clang-tidy/tool/CMakeLists.txt
@@ -17,6 +17,7 @@ target_link_libraries(clang-tidy
   clangTidyCERTModule
   clangTidyCppCoreGuidelinesModule
   clangTidyGoogleModule
+  clangTidyMISRAModule
   clangTidyLLVMModule
   clangTidyMiscModule
   clangTidyModernizeModule
diff --git a/clang-tidy/tool/ClangTidyMain.cpp b/clang-tidy/tool/ClangTidyMain.cpp
index b6684d02..05bd9144 100644
--- a/clang-tidy/tool/ClangTidyMain.cpp
+++ b/clang-tidy/tool/ClangTidyMain.cpp
@@ -493,6 +493,10 @@ extern volatile int SafetyModuleAnchorSource;
 static int LLVM_ATTRIBUTE_UNUSED SafetyModuleAnchorDestination =
     SafetyModuleAnchorSource;
 
+// This anchor is used to force the linker to link the MISRAModule.
+extern volatile int MISRAModuleAnchorSource;
+static int LLVM_ATTRIBUTE_UNUSED MISRAModuleAnchorDestination = MISRAModuleAnchorSource;
+
 } // namespace tidy
 } // namespace clang
 
